src/main/java/com/devature/penguin_api/config/SpringDataRestCustomization.java

```diff
@@ -0,0 +1,29 @@
+package com.devature.penguin_api.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.data.rest.core.config.RepositoryRestConfiguration;
+import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurer;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.config.annotation.CorsRegistry;
+
+// Spring Data REST CORS configuration
+@Component
```

Comment:
I think this should be a `@Configuration` annotation

Reply:
From the docs here https://docs.spring.io/spring-data/rest/docs/current-SNAPSHOT/reference/html/#reference I think this is correct, this is the example they document:

```@Component
public class CustomizedRestMvcConfiguration extends RepositoryRestConfigurer {

  @Override
  public void configureRepositoryRestConfiguration(RepositoryRestConfiguration config, CorsRegistry cors) {
    config.setBasePath("/api");
  }
}
```

Reply:
The docs state this step is unnecessary when using Spring Boot. https://docs.spring.io/spring-data/rest/docs/current-SNAPSHOT/reference/html/#getting-started.configuration

Reply:
From my experience today, unless there is an alternative way to configure CORS for the REST repositories, it is necessary

Reply:
You've already configured CORS [here](https://github.com/Devature/penguin-api/blob/3067b993380b66d8293bfc63361ad4cd4d38e27b/src/main/java/com/devature/penguin_api/PenguinApiApplication.java) though. I'm inclined to believe the documentation that this is unnecessary since I've never had to do this. Besides, the current production release is already working fine without it.

Reply:
The current production release does not have any REST repositories so it's not necessary. That global configuration is not used by the repositories, I did read that in some docs earlier but i don't have the source anymore. If you remove this you will find the CORS will work for the normal controller but not the REST repository endpoints

Reply:
I see what you mean. Well, I've already voiced my concerns and still think that this PR is unnecessary, but we can always remove this afterward if need be.